packages/react-html/README.md

```diff
@@ -12,112 +12,165 @@ $ yarn add @shopify/react-html
 
-**children**
-The children to be rendered inside the `#app` div.
+The function returns a pair of components:
 
-**styles**
-Descriptors for any style tags you want to include in the HEAD of the document.
+```tsx
+createSerializer<string>('MyData');
 
-**scripts**
-Descriptors for any script tags you want to include in your document. All scripts passed to this property will be deferred by appending them to the end of the document. We encourage this as a default, although you may use `blockingScripts` for any scripts that must be included in the HEAD of the document.
+// Would create the following components:
 
-**blockingScripts**
-Descriptors for any script tags you want to include in the HEAD of the document. These will block HTML parsing until they are evaluated, so use them carefully.
+function Serialize({data}: {data(): string}): null;
+function WithSerialized({
+  children,
+}: {
+  children(data: string | undefined): React.ReactNode;
+}): React.ReactNode;
+```
 
-**hideForInitialLoad**
-Sets the body contents to be hidden for the initial render. Use this when injecting stylesheets dynamically in development in order to prevent a flash of unstyled content.
+The general pattern for using these components is to render the `WithSerialized` component as the top-most child. Within the render prop, construct whatever stateful store/ manager you need, using the data that was retrieved in cases where the serialization was found (on the browser, or on subsequent server renders). Finally, render the UI that depends on that stateful part, and a `Serialize` component that extracts the part that you you need to communicate between server and client.
 
-### Serializers
+Here is a complete example, using `@shopify/react-i18n`’s support for async translations as the data that needs to be serialized:
 
-These props are useful for more complex applications that want to synchronize Redux, Apollo, translation, or any other data across the network boundary. These props are stringified into the DOM using (`@shopify/react-serialize`)[https://github.com/Shopify/quilt/blob/master/packages/react-serialize/README.md].
+```tsx
+import {createSerializer} from '@shopify/react-html';
+import {Provider, Manager} from '@shopify/react-i18n';
 
-**headData**
-Any serializable data that needs to be available from the DOM when the `synchronousScripts` are run.
+interface Props {
+  locale: string;
+  children: React.ReactNode;
+}
 
-**data**
-Any serializable data that needs to be available from the DOM when the `deferredScripts` are run.
+const {Serialize, WithSerialized} = createSerializer<
+  ReturnType<Manager['extract']>
+>('i18n');
+
+export default function I18n({locale, children}: Props) {
+  return (
+    <WithSerialized>
+      {translations => {
+        const manager = new Manager({locale}, translations);
+
+        return (
+          <>
+            <Provider manager={manager}>{children}</Provider>
+            <Serialize
+              data={() => ({
```

Comment:
Getting this error when linting the code:
> JSX props should not use arrow functions [react/jsx-no-bind]

Reply:
We'll be turning this rule off some point soon, most developers learning React see (and want to use) arrow functions in JSX all over online examples/ docs.